contracts/NodeDelegatorETH.sol

```diff
@@ -0,0 +1,411 @@
+
+import { IAccessControl } from "@openzeppelin/contracts/access/IAccessControl.sol";
+import { IERC20 } from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import { PausableUpgradeable } from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+import { UtilLib } from "./utils/UtilLib.sol";
+import { LRTConstants } from "./utils/LRTConstants.sol";
+import { LRTConfigRoleChecker, ILRTConfig } from "./utils/LRTConfigRoleChecker.sol";
+
+import { IDelegationManager } from "./eigen/interfaces/IDelegationManager.sol";
+import { IDelayedWithdrawalRouter } from "./eigen/interfaces/IDelayedWithdrawalRouter.sol";
+import { IEigenPodManager, IEigenPod } from "./eigen/interfaces/IEigenPodManager.sol";
+import { ISignatureUtils } from "./eigen/interfaces/ISignatureUtils.sol";
+import { IStrategyManager, IStrategy } from "./eigen/interfaces/IStrategyManager.sol";
+import { INodeDelegatorETH } from "./interfaces/INodeDelegatorETH.sol";
+import { ISSVNetwork, Cluster } from "./interfaces/ISSVNetwork.sol";
+// import { IOETH } from "./interfaces/IOETH.sol";
+import { IWETH } from "./interfaces/IWETH.sol";
+
+struct ValidatorStakeData {
+    bytes pubkey;
+    bytes signature;
+    bytes32 depositDataRoot;
+}
+
+/// @title NodeDelegatorETH Contract
+/// @notice The contract that handles the depositing WETH into SSV validators
+contract NodeDelegatorETH is
+    INodeDelegatorETH,
+    LRTConfigRoleChecker,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    address public constant DEAD_ADDRESS = 0x000000000000000000000000000000000000dEaD;
+    /// @dev The Wrapped ETH (WETH) contract address with interface IWETH
+    address public immutable WETH;
+
+    /// @dev The EigenPod is created and owned by this contract
+    address public eigenPod;
+    /// @dev Tracks the balance staked to validators and has yet to have the credentials verified with EigenLayer.
+    /// call verifyWithdrawalCredentials to verify the validator credentials on EigenLayer
+    uint256 public stakedButNotVerifiedEth;
+
+    uint256 internal constant DUST_AMOUNT = 10;
+    uint256 public constant FULL_STAKE = 32 ether;
+    uint256 public constant MIN_SLASHED_AMOUNT = 26 ether;
+
+    mapping(bytes32 pubkeyHash => bool hasStaked) public validatorsStaked;
+
+    /// @dev Maps the withdrawalRoots from the EigenLayer DelegationManager to the staker requesting the withdrawal.
+    /// Is not populated for internal withdrawals
+    mapping(bytes32 => address) public withdrawalRequests;
+    /// @dev Maps each EigenLayer strategy to the total amount of shares pending from internal withdrawals.
+    /// This does not include pending external withdrawals from Stakers as the PrimeETH total supply
+    // is reduced on external withdrawal request.
+    mapping(address => uint256) public pendingInternalShareWithdrawals;
```

Comment:
Should this be marked deprecated in this contract? It's never used in the code, outside of undelegate(), but I think that portion of undelegate() should be deleted.